source/include/core_mqtt_config_defaults.h

```diff
@@ -106,11 +106,11 @@
  * is irrelevant to the behavior of the library.
  *
  * <b>Possible values:</b> Any positive integer up to SIZE_MAX. <br>
- * <b>Default value:</b> `500`
+ * <b>Default value:</b> `5000`
  */
 #ifndef MQTT_PINGRESP_TIMEOUT_MS
-    /* Wait 0.5 seconds by default for a ping response. */
-    #define MQTT_PINGRESP_TIMEOUT_MS    ( 500U )
+    /* Wait 5 seconds by default for a ping response. */
+    #define MQTT_PINGRESP_TIMEOUT_MS    ( 5000U )
```

Comment:
I thought 5 seconds was a more reasonable default for this, for slower network conditions. From just testing locally with QEMU, it takes > 1 second for a CONNACK to be received, and sometimes with device advisor, the CONNACK takes 5-10 seconds to arrive. Since this bugfix will make this macro actually be used, we might be getting more bug reports from customers if we don't update this value now too.